Objective-C/CBLArray.mm

```diff
@@ -281,4 +312,15 @@ - (NSUInteger) hash {
 }
 
 
+#pragma mark - Lock
+
+
+- (NSObject*) sharedLock {
+    CBLDatabase* db = nil;
```

Comment:
Precompute this in the init method and store it in an ivar; it's too expensive to compute on every property access. (Probably needs to be marked as `__weak` to avoid a ref-cycle.)